tools/pot/openvino/tools/pot/algorithms/quantization/fake_quantize.py

```diff
@@ -386,32 +386,43 @@ def broadcast_fq_values(fq, node, min_level, max_level, fq_config):
     return min_level, max_level
 
 
-def set_rescaling_factors(target_device, model, scaling_factor=2.0):
+def set_rescaling_factors(config, model, scaling_factor=2.0):
     """
         Generate a list of weight FQ nodes for input convolutions
         for further rescaling of weights/FQs.
         Skip if target device is not CPU.
-        :param target_device: target device name
+        :param config: algo config
         :param model: NXModel instance
         :param scaling_factor: rescaling factor for first convolution nodes
     """
+
     fqs_to_rescale = []
+    saturation_fix = config.get('saturation_fix', 'first_layer')
 
-    if target_device not in ['CPU', 'ANY'] or not get_nodes_by_type(model, ['Convolution'], recursively=False):
+    if config['target_device'] not in ['CPU', 'ANY'] \
+            or not get_nodes_by_type(model, ['Convolution'], recursively=False) \
```

Comment:
Why it is for `Convolution` only but not for `OPERATIONS_WITH_WEIGHTS`?

Reply:
`get_first_convolutions` returns only convolutions, it's true. Should we extend `get_first_convolutions -> get_first_fully_connected` with the `OPERATIONS_WITH_WEIGHTS`, not only Convolutions?

Reply:
I think no. Ok, let's keep it as is at least for this release. It should be safe.